### ISSUE TITLE
Cannot change station

### ISSUE THREAD

**Issue by cbiere** [y20k/transistor#291]
Installed version: 4.0.4
I don't know whether it's a new bug but I've only noticed it now and can reproduce it on different android devices with different versions of Android. After adding about 23 stations, trying to switch to any other station will fail and playback jumps back to the current station.

**Comment by guterschwimmer** [comment]
Installed version 4.0.5
Same behaviour. I cannot play different stations after several uses. Initially works. Deleted that station that was played, it plays the first station in the list, deleted that, new first station was played.

**Comment by y20k** [comment]
Hi there. Sorry for the late answer. I am on vacation and cannot really look into the problem at the moment.

If there is a reliable way to reproduce the problem, please post instructions on how to. That would be a great help, since I have not yet experienced the problem myself.

Also helpful would be, if not too much trouble:

- open the file `/Android/data/org.y20k.transistor/files/collection/collection.json` in a text editor
- copy the content of the above file in to a JSON formatter (for example https://jsonformatter.curiousconcept.com/)
- look for irregularities something like two stations having the same `uuid` or `streamUris`

As always when experimenting: I would advice to make a copy of the whole files folder (`/Android/data/org.y20k.transistor/files/`)

**Comment by cbiere** [comment]
Thank you for looking into this. Here's my collection.json: https://pastebin.com/2QmRbrd0
All the stations were added using the builtin search. I didn't add any URL manually. It's possible the issue only started after deleting a station as guterschwimmer mentioned. I'll try to reproduce it with as few steps as possible.

**Comment by cbiere** [comment]
I reduced it to two stations: https://pastebin.com/qgPKChj
The only way to switch to the second station is by swiping left and making the second a favorite. Apparently it will always play the topmost station.

**Comment by evilrhinohater** [comment]
I was experienced similar issues, albeit I was only to play the _last_ station in my list.

I wanted to try to make this issue reproducible, so I deleted all the stations and wanted to note in which order I added them. However, now I do no longer have this issue. I think the only difference is that I have upgraded to 4.0.5 (I think it was 4.0.4 before).

**Comment by y20k** [comment]
Thanks. For investigating this so thoroughly!! The steamContent issue is probably a good lead. I will look into this next month. Since I am currently still on vacation. :palm_tree:

**Comment by cbiere** [comment]
Yes, if I replace "unsupported" by "application/ogg" respectively "audio/aacp" Transistor behaves normally and I can change stations again.

**Comment by guterschwimmer** [comment]
I deleted all my stations again and added again from search. Now it seems to be working again. Cannot reproduce any more. - But I never played with favourites in the meantime as Christian reported

**Comment by guterschwimmer** [comment]
A moment before, I had this problem again. But perhaps a little different: I could not play the first station in list (after switching between several stations in my list).
Workaround: I deleted the not working station and saved it again. All fine again - for the moment

**Comment by y20k** [comment]
I am super sorry this problem exists. I still have no clue how to fix it. Anyone with a procedure to reproduce it is very welcome. Have a nice weekend, stay save

**Comment by mbomb007** [comment]
After removing all my stations and re-adding them, I don't have the issue. Also, is there a reason that stars are sometimes different colors?

**Comment by lukeallister** [comment]
I also experienced this bug today. Everything was working normally, then I did a settings > maintenance 'update station images' and 'update stations' and then it would only play the station I was already listening to. 

After deleting stations and re-adding them, the problem goes away.

**Comment by piskr** [comment]
That's how I solved the issue. Thank you

**Comment by mbomb007** [comment]
> I also experienced this bug today. Everything was working normally, then I did a settings > maintenance 'update station images' and 'update stations' and then it would only play the station I was already listening to.
> 
> After deleting stations and re-adding them, the problem goes away.

@y20k I did "update station images" first, and my stations worked fine, but after "update stations", the problem was back, so "update stations" causes the issue.

**Comment by y20k** [comment]
@mbomb007 
Thanks!!! I will investigate that.

**Comment by y20k** [comment]
Today I found the time (finally) to look into the issue with "Update Stations". I could reproduce the problem 100% with stations that were downloaded via the in app search feature. And the bug was easily found. Transistor did not run a content type detection when it was updating stations. As a result all stations were marked as content type "unsupported". That in turn caused the weird behavior.

**Comment by mbomb007** [comment]
Nice, thanks!

**Comment by Marian-Rusnak** [comment]
Is there any estimate **when this will be fixed**?
3 days, 3 months, or 3 years?
Asking whether I should just delete all stations and add them back manually now, or wait for the fix update.

**Comment by y20k** [comment]
I can publish an update with a fix for the Update Stations issue tomorrow. Not 100% sure if that will solve your issue, too. My guess would be that running Update Stations should fix your problem. I would be great if you could report back. But as you can see from the discussion above the cause for the bug was hard to find. I am still not completely sure if the fix will prevent it from happening again.

**Comment by Marian-Rusnak** [comment]
Update Stations does not fix the problem.

 

I did use the “update images” and “update stations” clicks some time before the problem developed.

 

Happy New year

 

Marian

**Comment by y20k** [comment]
I published a new release of Transistor, that fixes the Update Stations feature ([v4.0.10](https://github.com/y20k/transistor/releases/tag/v4.0.10)). A bug in this feature was known to cause the problems described in this issue. If anyone currently experiences this problem ("Cannot change station") try Update Stations with the new version. This should fix them.

As I said I am still not sure, if the bug in Update Stations was the only cause. If anyone finds a way to reproduce the problem, please write me the steps you took to re-create the problem and I will further investigate.

Thanks everyone here for your help and patience.

(and Happy 2021, btw.)

**Comment by Marian-Rusnak** [comment]
In the meantime, I uninstalled and installed anew (from Google Play store, version 4 0 9, you call this version “Andy Varhol” – who was born in the same region in me- what is today Eastern Slovakia).

I manually added the stations back. 

This works.

I think I will avoid the 2 update buttons.

Thanks

 

Marian

**Comment by y20k** [comment]
Hi @Marian-Rusnak, 

> you call this version “Andy Varhol” – who was born in the same region in me- what is today Eastern Slovakia

Greetings to Eastern Slovakia then. :slovakia: 

Regarding the naming scheme. Each major version of Transistor is named after a David Bowie song :guitar: :moon: The first release was [Young Americans](https://github.com/y20k/transistor/releases/tag/v0.9)
